src/spotify_to_tidal/sync.py

```diff
@@ -172,133 +172,159 @@ def get_tracks_from_spotify_playlist(spotify_session: spotipy.Spotify, spotify_p
-    repeat_on_request_error(sync_playlist, spotify_session, tidal_session, spotify_id, tidal_id, config)
-    results.append(tidal_id)
-  return results
+    asyncio.run(sync_playlist(spotify_session, tidal_session, spotify_playlist, tidal_playlist, config) )
 
 def pick_tidal_playlist_for_spotify_playlist(spotify_playlist, tidal_playlists: Mapping[str, tidalapi.Playlist]):
     if spotify_playlist['name'] in tidal_playlists:
       # if there's an existing tidal playlist with the name of the current playlist then use that
       tidal_playlist = tidal_playlists[spotify_playlist['name']]
-      return (spotify_playlist['id'], tidal_playlist.id)
+      return (spotify_playlist, tidal_playlist)
     else:
-      return (spotify_playlist['id'], None)
-    
+      return (spotify_playlist, None)
 
 def get_user_playlist_mappings(spotify_session: spotipy.Spotify, tidal_session: tidalapi.Session, config):
-  results = []
-  spotify_playlists = get_playlists_from_spotify(spotify_session, config)
-  tidal_playlists = get_tidal_playlists_dict(tidal_session)
-  for spotify_playlist in spotify_playlists:
-      results.append( pick_tidal_playlist_for_spotify_playlist(spotify_playlist, tidal_playlists) )
-  return results
-
-def get_playlists_from_spotify(spotify_session: spotipy.Spotify, config):
+    results = []
+    spotify_playlists = asyncio.run(get_playlists_from_spotify(spotify_session, config))
+    tidal_playlists = get_tidal_playlists_dict(tidal_session)
+    for spotify_playlist in spotify_playlists:
+        results.append( pick_tidal_playlist_for_spotify_playlist(spotify_playlist, tidal_playlists) )
+    return results
+
+async def get_playlists_from_spotify(spotify_session: spotipy.Spotify, config):
     # get all the user playlists from the Spotify account
     playlists = []
-    with tqdm(total=1.0) as pbar:
-      pbar.set_description("Loading Spotify playlists")
-      spotify_results = spotify_session.user_playlists(config['spotify']['username'])
-      total = spotify_results['total']
-      exclude_list = set([x.split(':')[-1] for x in config.get('excluded_playlists', [])])
-      while True:
-          pbar.update(len(spotify_results['items'])/total)
-          for spotify_playlist in spotify_results['items']:
-              if spotify_playlist['owner']['id'] == config['spotify']['username'] and not spotify_playlist['id'] in exclude_list:
-                  playlists.append(spotify_playlist)
-          # move to the next page of results if there are still playlists remaining
-          if spotify_results['next']:
-              spotify_results = spotify_session.next(spotify_results)
-          else:
-              break
+    print("Loading Spotify playlists")
+    results = spotify_session.user_playlists(config['spotify']['username'])
+    exclude_list = set([x.split(':')[-1] for x in config.get('excluded_playlists', [])])
+      
+    # get all the remaining playlists in parallel
+    if results['next']:
```

Comment:
After the first chunk of playlists is received, any remaining chunks are fetched concurrently